File: infrastructure/services/application-service/database/migrate.js
'use strict'

require('dotenv').config({ path: require('path').join(__dirname, '..', '.env') })
const fs = require('fs')
const path = require('path')
const mysql = require('mysql2/promise')

async function main () {
  const host = process.env.DB_HOST || '127.0.0.1'
  const port = Number(process.env.DB_PORT) || 3306
  const user = process.env.DB_USER || 'root'
  const password = process.env.DB_PASSWORD
  const database = process.env.DB_NAME || 'application_db'

  const conn = await mysql.createConnection({
    host,
    port,
    user,
    password,
    multipleStatements: true
  })

  const schema = fs.readFileSync(path.join(__dirname, 'schema.sql'), 'utf8')
  await conn.query(schema)
  await conn.end()
  console.log(`Schema applied to ${database}`)
}

main().catch((err) => {
  console.error(err)
  process.exit(1)
})

File: services/application-service/backend/app/migrate.py
import sys
from pathlib import Path

import pymysql
from pymysql.constants import CLIENT

from app.core.config import settings


def _schema_path() -> Path:
    """Docker: /app/app → here.parent/database. Local: backend/app → parent.parent/database."""
    here = Path(__file__).resolve().parent
    for base in (here.parent / "database", here.parent.parent / "database"):
        p = base / "schema.sql"
        if p.is_file():
            return p
    raise FileNotFoundError(f"database/schema.sql not found near {here}")


def main() -> None:
    schema_path = _schema_path()
    sql = schema_path.read_text(encoding="utf-8")
    conn = pymysql.connect(
        host=settings.db_host,
        port=settings.db_port,
        user=settings.db_user,
        password=settings.db_password,
        charset="utf8mb4",
        client_flag=CLIENT.MULTI_STATEMENTS,
    )
    try:
        with conn.cursor() as cur:
            cur.execute(sql)
        conn.commit()
        print(f"Schema applied from {schema_path}", file=sys.stderr)
    finally:
        conn.close()


if __name__ == "__main__":
    main()
